### PR TITLE
A-RexCodeQuality

### PR DESCRIPTION
**Pull request by rexkoh425**
improve error handling

### CODE REVIEW

**Comment by AaronZZ10** [comment]
LGTM!!

File: docs/UML/AddCustomerSequence.puml [hunk 1]
```diff
@@ -67,4 +67,5 @@ activate CustomerList
     Parser <-- CustomerList : return
 deactivate CustomerList
 deactivate Parser
+hide footbox
```
Comment (CT9ARyan) [inline docs/UML/AddCustomerSequence.puml:70]: Great!! 👍

File: src/main/java/file/CustomerFile.java [hunk 2]
```diff
@@ -100,8 +99,12 @@ public void updateCustomerDataFile() throws IOException {
      */
     public void addCustomerWithParameters(String[] parameters, ArrayList<Integer> errorLines , int line) {
         assert parameters.length == Customer.NUMBER_OF_PARAMETERS : "wrong no. of parameter";
-        String customerName = parameters[0];
+
         try {
+            if(FileHandler.containEmptyParameter(parameters)){
```
Comment (CT9ARyan) [inline src/main/java/file/CustomerFile.java:104]: Maybe you could add a space here after the `if` and the closing `)` bracket.

File: src/main/java/file/FileHandler.java [hunk 3]
```diff
@@ -99,6 +100,15 @@ private static void createFolder() {
         }
     }
 
+    public static boolean containEmptyParameter(String[] parameters){
+        for(int i = 0; i < parameters.length ; i++){
```
Comment (CT9ARyan) [inline src/main/java/file/FileHandler.java:104]: Same thing here maybe?? Add a space after the `for`??

File: src/main/java/file/TransactionFile.java [hunk 4]
```diff
@@ -121,6 +126,7 @@ public void addTransactionWithParameters(String[] parameters , ArrayList<Integer
             boolean isCompleted = Boolean.parseBoolean(parameters[5]);
             Transaction transaction = new Transaction(transactionId , carLicensePlate, borrowerName, duration,
                     startDate, isCompleted);
+
             TransactionList.addTxWithoutPrintingInfo(transaction);
         }catch (NumberFormatException | DateTimeParseException | CarException | TransactionException e){
```
Comment (CT9ARyan) [inline src/main/java/file/TransactionFile.java:131]: Maybe leave a space here before `catch`?

File: src/main/java/file/FileHandler.java [hunk 5]
```diff
@@ -99,6 +100,15 @@ private static void createFolder() {
...
+            if(parameters[i].trim().isEmpty()){
```
Comment (CT9ARyan) [inline src/main/java/file/FileHandler.java:105]: Same thing here?

File: src/main/java/file/TransactionFile.java [hunk 6]
```diff
@@ -57,10 +57,12 @@ public void loadTransactionData() throws FileNotFoundException, TransactionExcep
             Scanner scanner = new Scanner(transactionDataFile);
             ArrayList<Integer> errorLines = new ArrayList<>();
             int line = 1;
+
             while (scanner.hasNext()) {
                 scanLineAndAddTransaction(scanner, errorLines, line);
                 line ++;
             }
+
             if(!errorLines.isEmpty()) {
```
Comment (CT9ARyan) [inline src/main/java/file/TransactionFile.java:66]: Same thing here?

File: src/main/java/file/TransactionFile.java [hunk 7]
```diff
@@ -89,6 +91,7 @@ public void loadTransactionDataIfExist(){
     public void scanLineAndAddTransaction(Scanner scanner, ArrayList<Integer> errorLines, int line) {
         String input = scanner.nextLine();
         String[] parameters = input.split(" \\| ");
+
         if(parameters.length != Transaction.NUMBER_OF_PARAMETERS){
```
Comment (CT9ARyan) [inline src/main/java/file/TransactionFile.java:95]: Here as well?

File: src/main/java/file/TransactionFile.java [hunk 8]
```diff
@@ -109,9 +112,11 @@ public void addTransactionWithParameters(String[] parameters , ArrayList<Integer
         try {
 
             String transactionId = parameters[0];
-            if(!Transaction.isValidTxId(transactionId)){
+
+            if(!Transaction.isValidTxId(transactionId) || FileHandler.containEmptyParameter(parameters)){
```
Comment (CT9ARyan) [inline src/main/java/file/TransactionFile.java:116]: Maybe here too??

**Review by CT9ARyan — COMMENTED** [review]
Some indentation issues, otherwise LGTM!!👍

**Pull request merged**